README.md

```diff
@@ -1,53 +1,102 @@
+This is a mobile app project template using [Flutter](https://github.com/flutter) and the latest Flutter/Dart practices.
 
-1. Check out this repository
-2. Delete the `.git` folder
-3. Git init this repository and start working on your project!
-4. Prior to submitting your request for publication, make sure to review the
-   [Open Source guidelines for publications](https://nventive.visualstudio.com/Internal/_wiki/wikis/Internal_wiki?wikiVersion=GBwikiMaster&pagePath=%2FOpen%20Source%2FPublishing&pageId=7120).
+- It uses the MVVM pattern.
+- Code is organized by [application layer](doc/Architecture.md#Solution-Structure).
+- It comes with [dependency injection](doc/DependencyInjection.md).
+- There are built-in [logs](doc/Logging.md) and [diagnostic tools](doc/Diagnostics.md).
+- There is scaffolding code showing sample features.
+  When you run as-is, you get a _Dad Jokes_ application.
 
-## Features (to keep as-is, configure or remove)
-- [Mergify](https://mergify.io/) is configured. You can edit or remove [.mergify.yml](/.mergify.yml).
+## Preview
+From left to right: WinUI, iOS, and Android.
+![Platform-Comparison-Preview](https://user-images.githubusercontent.com/39710855/264705525-d070952c-04c7-4f4a-b6af-957f8415fb3e.png)
+> Note that scaling was adjusted to better fit this preview and therefore this image isn't representative of the actual device sizes.
 
-The following is the template for the final README.md file:
+### Diagnostic Tools
+![Diagnostics-Overlay-Preview](https://user-images.githubusercontent.com/39710855/264691340-dbc9d137-a199-4969-94d7-7dd430e08da7.gif)
 
----
+## Requirements
 
-# Project Title
+Visual Studio Code with Flutter are required.
 
-{Project tag line}
+This template largely relies on Flutter, if you want to make sure you got everything installed correctly on your machine, we encourage you to use `flutter doctor -v`, the documentation is available [here](https://docs.flutter.dev/)
 
-{Small description of the purpose of the project}
+## Getting Started
 
-[![License](https://img.shields.io/badge/License-Apache%202.0-blue.svg)](LICENSE)
+We use a custom PowerShell script and a Dart package to easily create new projects. It simplifies the **project renaming**.
 
-## Getting Started
+### Generate a new project
+
+1. Make sure you cloned the latest version of this repository.
+
+2. Make sure you have PowerShell installed.
+
+   > 💡 For Windows, you can run `powershell` to verify. You can install it with the command `winget install --id Microsoft.Powershell --source winget`.
+   > 💡 For Mac, you can run `pwsh` to verify. You can install it with Homebrew `brew install powershell/tap/powershell`.
+
+3. To run the script and create a new project, run the following command in the cloned `FlutterApplicationTemplate` repository.
+
+   `powershell -File ".\tools\copyApplicationTemplate.ps1" -sourceProjectDir C:\P\FlutterApplicationTemplate -destDir C:\P -projectName TODO -appName TODO -packageName TODO -organization Org`
```

Comment:
replace the TODO with example values and then explain what the example values represent